subprojects/core/src/main/java/org/gradle/internal/service/scopes/BasicGlobalScopeServices.java

```diff
@@ -103,8 +110,30 @@ DocumentationRegistry createDocumentationRegistry() {
     }
 
     @Provides
-    ExecFactory createExecFactory(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, ExecutorFactory executorFactory, TemporaryFileProvider temporaryFileProvider) {
-        return DefaultExecActionFactory.of(fileResolver, fileCollectionFactory, executorFactory, temporaryFileProvider);
+    PropertyFactory createPropertyFactory(PropertyHost propertyHost) {
+        return new DefaultPropertyFactory(propertyHost);
+    }
+
+    @Provides
+    DefaultFilePropertyFactory createFilePropertyFactory(PropertyHost propertyHost, FileResolver fileResolver, FileCollectionFactory fileCollectionFactory) {
+        return new DefaultFilePropertyFactory(propertyHost, fileResolver, fileCollectionFactory);
+    }
+
+    @Provides
+    ExecObjectFactory createExecObjectFactory(FileCollectionFactory fileCollectionFactory, PropertyFactory propertyFactory, FilePropertyFactory filePropertyFactory) {
+        // In BasicGlobalScopeServices, we can't access all the services we use direct instantiator for now
+        return new DefaultExecObjectFactory(DirectInstantiator.INSTANCE, fileCollectionFactory, propertyFactory, filePropertyFactory);
```

Comment:
💭 Before we used `InstantiatorBackedObjectFactory` somewhere lower that used DirectInstantiator. Now this is brought up to services class so we have more control